[Frontendv2]Added button to allow show or hide password in profile

Alabhya268 opened the pull request:
Fixes #3277

Currently in Frontendv2 when changing password from profile there is not button to make password show or hidden.
This PR fixes this issue.

Because Frontendv2 uses same template for other forms, I had to make necessary change specifically for password field.

Alabhya268 commented:
https://user-images.githubusercontent.com/57143358/110207292-92f3f880-7ea8-11eb-8005-b02c08991c6b.mp4

@Ram81 @RishabhJain2018 , Please review this PR.

Alabhya268 commented:
@Ram81 , Please have a look at this PR.

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.ts:
```diff
@@ -75,6 +75,11 @@ export class ModalComponent implements OnInit {
+  /**
+   * password feild visibility
+   */
+  passwordVisibility:String = 'password';
```
@Alabhya268 instead of storing text `password` or `text` can we switch to a boolean `showPassword`

Alabhya268 replied:
ok

Alabhya268 commented:
@Ram81 , I am now made variable boolean.

I was using variable previously to change visibility of password which was resulting in simultaneously changing visibilty of all password field. I have now fixed that by making different variable for different password field.

Let me know if anything else needs to be changed.

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.ts:
```diff
@@ -36,9 +36,6 @@ export class ModalComponent implements OnInit {
   /**
    * Invalid fields
-   */
-  invalidFields = [];
```
@Alabhya268 why are we removing this field

Alabhya268 replied:
I may have done it by mistake. I'll fix it instantly.

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.ts:
```diff
@@ -36,9 +36,6 @@ export class ModalComponent implements OnInit {
    * Invalid fields
-   */
-  invalidFields = [];
-  invalidFieldsAsText = '';
```
same here

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.ts:
```diff
@@ -75,6 +72,21 @@ export class ModalComponent implements OnInit {
+  /**
+   * old_password field visibility
+   */
+  old_passwordVisibility:boolean = false;
```
Please use consistent naming convention

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.ts:
```diff
@@ -75,6 +72,21 @@ export class ModalComponent implements OnInit {
+  /**
+   * new_password1 field visibility
+   */
+   new_password1Visibility:boolean = false;
```
Change to `newPasswordTextVisibility`

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.ts:
```diff
@@ -75,6 +72,21 @@ export class ModalComponent implements OnInit {
+   /**
+   * new_password2 field visibility
+   */
+  new_password2Visibility:boolean = false;
```
Change to `reenteredPasswordTextVisibility`

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.ts:
```diff
@@ -334,4 +346,5 @@ export class ModalComponent implements OnInit {
   validateFileInput(e) {
     this.isDisabled = e.target.value === '';
   }
+  
```
remove newline

Ram81 reviewed (changes requested):
@Alabhya268 the PR is not compiling. Can you please make the necessary fixes.

Alabhya268 commented:
Ok I will make these changes

Alabhya268 commented:
@Ram81 , I've now change the names according to suggestions. I have also fixed compiling issues. Let me know if anything else needs to be changed.

Alabhya268 commented:
@Ram81 , Please review this PR.

Alabhya268 commented:
@Ram81, Please also review this PR again.

Alabhya268 commented:
@Ram81 , ping for PR.

Alabhya268 commented:
@Ram81 please look at this PR again.

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.ts:
```diff
@@ -75,6 +75,21 @@ export class ModalComponent implements OnInit {
+  /**
+   * old_password field visibility
+   */
+  oldpasswordTextVisibility:boolean = false;
```
Fix case inconsitency

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.ts:
```diff
@@ -75,6 +75,21 @@ export class ModalComponent implements OnInit {
+   /**
+   * new_password2 field visibility
+   */
+  reenteredPasswordTextVisibility:boolean = false;
```
change to `reEnteredPasswordTextVisibility`

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.ts:
```diff
@@ -75,6 +75,21 @@ export class ModalComponent implements OnInit {
+  newPasswordTextVisibility:boolean = false;
+
+   /**
+   * new_password2 field visibility
```
fix the comment

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.ts:
```diff
@@ -75,6 +75,21 @@ export class ModalComponent implements OnInit {
+  oldpasswordTextVisibility:boolean = false;
+
+  /**
+   * new_password1 field visibility
```
fix the comment

Ram81 commented on frontend_v2/src/app/components/utility/modal/modal.component.html:
```diff
@@ -16,19 +16,92 @@
+                  role="button"
+                  tabindex="0"
+                >
+                  <i *ngIf = "!reenteredPasswordTextVisibility" class="fa fa-eye pointer ng-scope"></i>
```
rename variable

Alabhya268 commented:
@Ram81 , I have made changes needed .

Alabhya268 commented:
@Ram81 ,I've made changes needed, Please have a look at this PR again.

Alabhya268 commented:
@Ram81, Please have a look at this PR again.

Alabhya268 commented:
@Ram81 Ping for this PR.

Alabhya268 commented:
@Ram81, Could you please review this PR.

Alabhya268 commented:
@Ram81, I have renamed the variables and fixed the comment, Please have a look at this PR.

Alabhya268 commented:
@Ram81, I have renamed the variables and fixed the comment, Please have a look at this PR. This PR has been open for a long time, Please give feedback.

Alabhya268 commented:
https://user-images.githubusercontent.com/57143358/112752822-79028d00-8ff2-11eb-880f-6a09e4e224ca.mp4

Alabhya268 commented:
@Ram81, Is there something wrong with PR. Please give a feedback

Ram81 commented:
@Alabhya268 this change is not a high priority change. I'll take a look at the PR sometime this week meanwhile you can work on other issues.

Alabhya268 commented:
@Ram81, Please review this PR.

Alabhya268 commented:
> @Alabhya268 this change is not a high priority change. I'll take a look at the PR sometime this week meanwhile you can work on other issues.

@Ram81, did you check this PR ?

Alabhya268 commented:
@Ram81, Please have a look at this PR.
